shm_wrapper/shm_wrapper.c

```diff
@@ -278,26 +265,31 @@ void shm_init (process_t process)
 		
 		//mutual exclusion semaphore, catalog_mutex with initial value = 0
 		if ((catalog_sem = sem_open(CATALOG_MUTEX_NAME, O_CREAT, 0660, 0)) == SEM_FAILED) {
-			error("sem_open: catalog_mutex@dev_handler");
+			log_printf(FATAL, "sem_open: catalog_mutex@dev_handler");
```

Comment:
For some of these, I see you don't include the `strerror(errno)` so just make sure you add those to all the system C functions.

Reply:
👌